### ISSUE TITLE
Implement Git submodules to track dependencies

### ISSUE THREAD

**Issue by TimothyEDawson** [EdwardALuke/loci#34]
This issue implements Git submodule functionality in order to begin tracking third-party dependencies of the Loci solver, both mandatory and optional.

The current scope is limited to pinning relatively recent releases of each library which have been vetted and are known to work with Loci. In the future this infrastructure could be further developed to support semi-automated dependency installation and patches to the external software if ever deemed necessary.

The software to be tracked currently includes:
- [HDF5](https://github.com/HDFGroup/hdf5) v1.14.5
- [GKlib](https://github.com/KarypisLab/GKlib) + [METIS](https://github.com/KarypisLab/METIS) v5.1.0 + [ParMETIS](https://github.com/KarypisLab/ParMETIS) v4.0.3
- [PT-Scotch](https://gitlab.inria.fr/scotch/scotch) v7.0.4
- [PETSc](https://github.com/petsc/petsc) v3.18.5

Additional dependencies which are not being tracked include:
- [openmpi](https://github.com/open-mpi/ompi) v4.1.5
- [libxml2](https://github.com/GNOME/libxml2) v2.13.5 (unvetted)
- [CGNS](https://github.com/CGNS/CGNS) v4.4.0 (unvetted)

**Comment by EdwardALuke** [comment]
The scotch that I have verified is v7.0.4.  One issue that I have identified is that it two subsequent runs on the same number of processors does not produce the same partition.  This results in chem failing restart quicktests because they don't converge the solution very far and the linear system solver error is large enough to cause a failure.  I have tried to set the random number seed to prevent this behavior, but so far I haven't been able to get an interface to ptscotch that doesn't behave this way.  In general this isn't a problem, but I think most users will expect that the partition will be the same if run on the same cluster with the same number of processors.

I imagine 7.0.5 would also work, but I have not tested it.
